src/os/unix.rs

```diff
@@ -142,8 +144,22 @@ impl WindowExt for Window {
 
 /// Additional methods on `WindowBuilder` that are specific to Unix.
 pub trait WindowBuilderExt {
-
+    fn with_visual(self, visual_infos: *const XVisualInfo) -> WindowBuilder;
```

Comment:
Please do `with_visual<P>(self, visual_infos: *const P) -> WindowBuilder` instead, to avoid having to expose types from X11.